x-pack/plugins/maps/public/connected_components/mb_map/draw_control/draw_control.tsx

```diff
@@ -13,37 +13,55 @@ import MapboxDraw from '@mapbox/mapbox-gl-draw';
 import DrawRectangle from 'mapbox-gl-draw-rectangle-mode';
 import type { Map as MbMap } from '@kbn/mapbox-gl';
 import { Feature } from 'geojson';
-import { DRAW_TYPE } from '../../../../common/constants';
+import { DRAW_SHAPE } from '../../../../common/constants';
 import { DrawCircle } from './draw_circle';
 import { DrawTooltip } from './draw_tooltip';
 
+const SIMPLE_SELECT = 'simple_select';
 const DRAW_RECTANGLE = 'draw_rectangle';
 const DRAW_CIRCLE = 'draw_circle';
+const DRAW_POLYGON = 'draw_polygon';
+const DRAW_LINE = 'draw_line_string';
+const DRAW_POINT = 'draw_point';
+
+const mbModeEquivalencies = new Map([
+  ['simple_select', 'SIMPLE_SELECT'],
+  ['draw_rectangle', 'BOUNDS'],
+  ['draw_circle', 'DISTANCE'],
+  ['draw_polygon', 'POLYGON'],
+  ['draw_line_string', 'LINE'],
+  ['draw_point', 'POINT'],
+]);
 
 const mbDrawModes = MapboxDraw.modes;
 mbDrawModes[DRAW_RECTANGLE] = DrawRectangle;
 mbDrawModes[DRAW_CIRCLE] = DrawCircle;
 
 export interface Props {
-  drawType?: DRAW_TYPE;
-  onDraw: (event: { features: Feature[] }) => void;
+  drawShape?: DRAW_SHAPE;
+  onDraw: (drawControl: MapboxDraw) => (event: { features: Feature[] }) => void;
```

Comment:
This is a hard to follow signature that can be avoid by adding a wrapper around props.onDraw to DrawControl. That wrapper can then call props.onDraw and include drawControl as a parameter.

So keep signature as `onDraw: (event: { features: Feature[] }) => void;`

Then add the function DrawControl._onDraw like so and remove the double function signature. When props.onDraw is called, it will have everything it needs.
```
_onDraw = (event: { features: Feature[] }) => {
  this.props.onDraw(drawControl, event);
}
```